### PR TITLE
Update package name input and access scope check 

### PR DESCRIPTION
**Pull request by elsh**
Allow Swift as a package name
Update AccessScope::isChildOf
Add more tests for package access level
Resolves rdar://106728606

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
This PR addresses comments in https://github.com/apple/swift/pull/64230 as well as adds more tests. Merging this PR but feel free to leave any comments and they will be addressed in the upcoming PRs.

File: test/attr/accessibility_proto.swift [hunk 1]
```diff
@@ -1,91 +1,202 @@
-// RUN: %target-typecheck-verify-swift
-// RUN: %target-swift-frontend -typecheck -disable-access-control %s
+// RUN: %target-swift-frontend -typecheck -disable-access-control -package-name myPkg %s
```
Comment (xymus) [inline test/attr/accessibility_proto.swift:1]: It looks like this used to test both the diagnostics and the disabling of access checking. I believe this dropped the check on the diagnostic. Can we keep a -verify check?

**Review by xymus — APPROVED** [review]
It's worth fixing the test, but besides that this looks good!

**Comment by nkcsgexi** [comment]
Thank you for testing this extensively, Ellie!

File: test/attr/accessibility_proto.swift (hunk 1 above)
Comment (elsh) [reply test/attr/accessibility_proto.swift:1]: Added it back in https://github.com/apple/swift/pull/64408

**Pull request merged**